### DISCUSSION TITLE
How to separate push events in Actions?

### DISCUSSION THREAD

**Discussion by adslbarxatov** [community/community#26412] [Actions]
<p>Hi!</p>
<p>Is there some method, how to separate <code>push</code> event, that creates a file in repo, from <code>push</code> event, that removes or changes (at least, removes) it?</p>
<p>Docs say that this event ignores (am I right?) <code>types</code>. But we already tried <code>paths</code> (like <code>Release/*.exe</code>), and <code>push</code> event reacts on them correctly.</p>
<p>Thanks for answers!</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/adslbarxatov">@adslbarxatov</a>,</p>
<p>Yes, the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#push" rel="noopener nofollow ugc"><strong>push</strong></a> event does not have optional activity types.</p>
<p>You can use the path filters to configure a workflow to run only when there is any modified file can match the patterns you specified.<br>
Any file that is <strong>added</strong>, <strong>updated</strong> or <strong>deleted</strong> in the pushed commits can let the push event trigger the workflow.</p>
<p>If you really need the <strong>push</strong> event can support some activity types, I recommend that you can directly share a feature request in the <a href="https://support.github.com/contact/feedback" rel="noopener nofollow ugc"><strong>Feedback form for GitHub</strong></a>.<br>
That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by Simran-B** [comment]
<p>As a workaround, you could use <code>actions/checkout@v2</code>, probably with <code>depth: 0</code>, and use Git to determine whether files were added, modified or deleted. Based on that information, you can skip individual jobs.</p>

**Comment by adslbarxatov** [comment]
<p>Thank you!</p>
<p>Looks like this method allows us to make events triggering more flexible and accurate. But it’s still requires additional preparations. So, we think that this question can get simpler answer in the future</p>

**Comment by adslbarxatov** [comment]
<p>Thank you!</p>
<p>You’re right: trigger file is the simplest way to emulate needed event conditions. But anyway we’ll offer <code>push</code> types in order to solve this using only <code>.yaml</code> file (without anything else)</p>
